src/test/java/com/duckblade/osrs/dpscalc/calc/gearbonus/VampyreGearBonusTest.java

```diff
@@ -1,38 +1,82 @@
 import com.duckblade.osrs.dpscalc.calc.compute.ComputeInputs;
-import com.duckblade.osrs.dpscalc.calc.testutil.DefenderAttributesUtil;
+import com.duckblade.osrs.dpscalc.calc.model.AttackType;
 import com.duckblade.osrs.dpscalc.calc.model.DefenderAttributes;
 import com.duckblade.osrs.dpscalc.calc.model.GearBonuses;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import com.duckblade.osrs.dpscalc.calc.testutil.DefenderAttributesUtil;
+import net.runelite.api.ItemID;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
 import org.mockito.junit.jupiter.MockitoExtension;
 
+import static com.duckblade.osrs.dpscalc.calc.testutil.AttackStyleUtil.ofAttackType;
+import static com.duckblade.osrs.dpscalc.calc.testutil.DefenderAttributesUtil.VAMPYRE;
+import static com.duckblade.osrs.dpscalc.calc.testutil.ItemStatsUtil.ofItemId;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
 @ExtendWith(MockitoExtension.class)
 class VampyreGearBonusTest
 {
+	@Mock
+	private WeaponComputable weaponComputable;
 
 	@Mock
 	private ComputeContext context;
 
 	@InjectMocks
 	private VampyreGearBonus vampyreGearBonus;
 
+	@Test
+	void isApplicableWhenUsingVampyrebaneAgainstVampyres()
+	{
+		when(context.get(ComputeInputs.DEFENDER_ATTRIBUTES)).thenReturn(DefenderAttributesUtil.VAMPYRE);
+		when(context.get(ComputeInputs.ATTACK_STYLE)).thenReturn(ofAttackType(AttackType.CRUSH));
+		when(context.get(weaponComputable)).thenReturn(
+				ofItemId(ItemID.IVANDIS_FLAIL),
+				ofItemId(ItemID.BLISTERWOOD_SICKLE),
+				ofItemId(ItemID.BLISTERWOOD_FLAIL)
+		);
+
+		assertTrue(vampyreGearBonus.isApplicable(context));
+		assertTrue(vampyreGearBonus.isApplicable(context));
+		assertTrue(vampyreGearBonus.isApplicable(context));
+	}
+
+	@Test
+	void isNotApplicableForMeleeWithWrongWeapon()
+	{
+		when(context.get(ComputeInputs.DEFENDER_ATTRIBUTES)).thenReturn(DefenderAttributesUtil.VAMPYRE);
+		when(context.get(ComputeInputs.ATTACK_STYLE)).thenReturn(ofAttackType(AttackType.STAB));
+		when(context.get(weaponComputable)).thenReturn(ofItemId(ItemID.SCYTHE_OF_VITUR));
+
+		assertFalse(vampyreGearBonus.isApplicable(context));
+	}
+
+	@Test
+	void isNotApplicableWhenNotUsingMelee()
+	{
+		when(context.get(ComputeInputs.DEFENDER_ATTRIBUTES)).thenReturn(DefenderAttributesUtil.VAMPYRE);
+		when(context.get(ComputeInputs.ATTACK_STYLE)).thenReturn(ofAttackType(AttackType.MAGIC));
+
+		assertFalse(vampyreGearBonus.isApplicable(context));
+	}
+
 	@Test
 	void isApplicableAgainstVampyres()
```

Comment:
This test case is redundant with `isApplicableWhenUsingVampyrebaneAgainstVampyres` and should be negated to `isNotApplicableAgainstNonVampyres`, since that test case is missing.

Reply:
Not 100% sure I understood you, but might have fixed it in the latest push